oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java

```diff
@@ -1055,13 +1064,20 @@ private <T extends Document> T findAndModify(Collection<T> collection,
                             Filters.eq(Document.MOD_COUNT, modCount)
                     );
 
-                    UpdateResult result = execute(session -> {
-                        if (session != null) {
-                            return dbCollection.updateOne(session, query, update);
-                        } else {
-                            return dbCollection.updateOne(query, update);
-                        }
-                    }, collection);
+                    UpdateResult result = null;
+                    try {
+                        result = execute(session -> {
+                            if (session != null) {
+                                return dbCollection.updateOne(session, query, update);
+                            } else {
+                                return dbCollection.updateOne(query, update);
+                            }
+                        }, collection);
+                    } catch (MongoWriteException e) {
+                        WriteError error = e.getError();
+                        LOG.warn(e.getMessage(), updateOp.getId());
```

Comment:
This would only work with placeholders `{}` not with normal strings.

This would print only the `e.getMessage()` and ignore the `updateOp.getId()` since it won't find any placeholder for this in `e.getMessage()`.